draft-ietf-tls-esni.md

```diff
@@ -215,8 +216,30 @@ SNI Encryption keys can be published using the following ESNIKeys structure.
         uint64 not_after;
         Extension extensions<0..2^16-1>;
     } ESNIKeys;
+
+    struct {
+        ESNIKeys esni_keys;
+        Extension dns_extensions<0..2^16-1>;
+    } ESNIRecord;
 ~~~~
 
+The outermost ESNIRecord structure contains the following fields:
+
+esni_keys
+: An ESNIKeys structure that contains the actual keys used to encrypt the SNI
+as well as some metadata related to those keys.
+
+dns_extensions
+: A list of extensions that the client can take into consideration when
+resolving the target DNS name. The format is defined in
+{{RFC8446}}; Section 4.2. The purpose of the field is to
+provide room for additional features in the future. An extension
+may be tagged as mandatory by using an extension type codepoint with
+the high order bit set to 1. A client which receives a mandatory extension
+they do not understand must reject the ESNIRecord values.
```

Comment:
Might be useful to call out that these extensions aren't verified as part of the record_digest?

Reply:
Agreed. I'll merge this and then make that change directly.